src/Microsoft.Extensions.Options/Options.cs

```diff
@@ -6,6 +6,30 @@
     public static class Options
     {
         /// <summary>
+        /// The order which will be used for the next call to Configure
+        /// </summary>
+        public static int DefaultConfigureOptionsOrder;
```

Comment:
Statics + the ability to specify the order explicitly can results into non-deterministic behavior if you create more than one container in the same process or AppDomain.